packages/flutter/lib/src/material/date.dart

```diff
@@ -8,38 +8,280 @@
 library;
 
 import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
 
 import 'material_localizations.dart';
 
+/// Controls the calendar system used in the date picker.
+///
+/// A [CalendarDelegate] defines how dates are interpreted, formatted, and
+/// navigated within the picker. Different calendar systems (e.g., Gregorian,
+/// Nepali, Hijri, Buddhist) can be supported by providing custom implementations.
+///
+/// {@tool dartpad}
+/// This example demonstrates how a [CalendarDelegate] is used to implement a
+/// custom calendar system in the date picker.
+///
+/// ** See code in examples/api/lib/material/date_picker/custom_calendar_date_picker.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [GregorianCalendarDelegate], the default implementation for the Gregorian calendar.
+///  * [CalendarDatePicker], which uses this delegate to manage calendar-specific behavior.
+@optionalTypeArgs
```

Comment:
Shall we remove this line and require the type arg? (Which agrees with https://github.com/flutter/flutter/pull/161874/files#r1980199306 )

Reply:
Removed